connector/src/main/java/org/geysermc/connector/network/translators/collision/translators/BlockCollision.java

```diff
@@ -95,44 +100,38 @@ public boolean correctPosition(GeyserSession session, BoundingBox playerCollisio
-
-                Vector3d southFacePos = Vector3d.from(b.getMiddleX(),
-                        b.getMiddleY(),
-                        b.getMiddleZ() + (b.getSizeZ() / 2));
-
-                Vector3d eastFacePos = Vector3d.from(b.getMiddleX()  + (b.getSizeX() / 2),
-                        b.getMiddleY(),
-                        b.getMiddleZ());
-
-                Vector3d westFacePos = Vector3d.from(b.getMiddleX()  - (b.getSizeX() / 2),
-                        b.getMiddleY(),
-                        b.getMiddleZ());
-
-                double translateDistance = northFacePos.getZ() - relativePlayerPosition.getZ() - (playerCollision.getSizeZ() / 2);
-                if (Math.abs(translateDistance) < CollisionManager.COLLISION_TOLERANCE * 1.1) {
+                double northFaceZPos = b.getMiddleZ() - (b.getSizeZ() / 2);
+                double translateDistance = northFaceZPos - relativePlayerPosition.getZ() - (playerCollision.getSizeZ() / 2);
+                if (Math.abs(translateDistance) < pushAwayTolerance) {
                     playerCollision.translate(0, 0, translateDistance);
                 }
-                
-                translateDistance = southFacePos.getZ() - relativePlayerPosition.getZ() + (playerCollision.getSizeZ() / 2);
-                if (Math.abs(translateDistance) < CollisionManager.COLLISION_TOLERANCE * 1.1) {
+
+                double southFaceZPos = b.getMiddleZ() + (b.getSizeZ() / 2);
+                translateDistance = southFaceZPos - relativePlayerPosition.getZ() + (playerCollision.getSizeZ() / 2);
+                if (Math.abs(translateDistance) < pushAwayTolerance) {
                     playerCollision.translate(0, 0, translateDistance);
                 }
 
-                translateDistance = eastFacePos.getX() - relativePlayerPosition.getX() + (playerCollision.getSizeX() / 2);
-                if (Math.abs(translateDistance) < CollisionManager.COLLISION_TOLERANCE * 1.1) {
+                double eastFaceXPos = b.getMiddleX() + (b.getSizeX() / 2);
+                translateDistance = eastFaceXPos - relativePlayerPosition.getX() + (playerCollision.getSizeX() / 2);
+                if (Math.abs(translateDistance) < pushAwayTolerance) {
                     playerCollision.translate(translateDistance, 0, 0);
                 }
 
-                translateDistance = westFacePos.getX() - relativePlayerPosition.getX() - (playerCollision.getSizeX() / 2);
-                if (Math.abs(translateDistance) < CollisionManager.COLLISION_TOLERANCE * 1.1) {
+                double westFaceXPos = b.getMiddleX() - (b.getSizeX() / 2);
+                translateDistance = westFaceXPos - relativePlayerPosition.getX() - (playerCollision.getSizeX() / 2);
+                if (Math.abs(translateDistance) < pushAwayTolerance) {
                     playerCollision.translate(translateDistance, 0, 0);
                 }
+
+                double bottomFaceYPos = b.getMiddleY() - (b.getSizeY() / 2);
```

Comment:
If there is a bottom face Y position collision check, should there not also be a top face check?

Reply:
I'm pretty sure that is handled in lines 83-92 as part of the step up check.